docs/dev/testing/sdk-integration-tests.md

```diff
@@ -1,9 +1,511 @@
+
+### Prerequisites
+
+Set the `LANGSMITH_API_KEY` environment variable:
+
+```bash
+export LANGSMITH_API_KEY="<your-api-key>"
+```
+
+### Run All Integration Tests
+
+```bash
+# Run all ignored (integration) tests
+cargo test --test integration_test -- --ignored --nocapture
+
+# Run specific integration test
+cargo test --test integration_test test_list_prompts_from_prompthub -- --ignored --nocapture
+```
+
+### Run Unit Tests Only (Default)
+
+```bash
+# This skips integration tests (they're ignored by default)
+cargo test
+```
+
+### Run Everything
+
+```bash
+# Run both unit and integration tests
+cargo test -- --include-ignored
+```
+
+## Available Integration Tests
+
+### PromptHub Tests
+
+#### `test_list_prompts_from_prompthub` ✅
+
+**Status**: Working
+
+Tests the ability to list prompts from the LangSmith PromptHub.
+
+**Requirements**:
+- Valid `LANGSMITH_API_KEY`
+- Read permissions
+
+**What it tests**:
+- Authentication with LangSmith API
+- Fetching and parsing paginated prompt list
+- API response deserialization
+
+#### `test_push_prompt_to_prompthub` ✅
+
+**Status**: Working (requires prompt repository to exist first)
+
+Creates a new commit for a prompt in the LangSmith PromptHub.
+
+**Prerequisites**:
+1. Valid `LANGSMITH_API_KEY` with **write permissions**
+2. The prompt repository must already exist in your PromptHub
+   - Create at: https://smith.langchain.com/prompts
+   - Test uses: `codekiln/langstar-integration-test`
+
+**What it tests**:
+- Fetching current organization information
+- Setting X-Organization-Id header for write operations
+- Creating a commit using `POST /api/v1/commits/{owner}/{repo}`
+- Proper request body format (manifest, parent_commit, example_run_ids)
+
+**Expected behavior**:
+- If repository exists: ✅ Returns commit hash
+- If repository doesn't exist: ❌ Returns 404 "Repository not found"
+
+### Assistant Tests
+
+#### Prerequisites
+
+**1. Deploy Test LangGraph Application**
+
+Before running assistant integration tests, you must deploy the test graph:
+
+1. Follow the deployment guide: `../../tests/fixtures/test-graph-deployment/DEPLOYMENT_GUIDE.md`
+2. Deploy to LangGraph Cloud via LangSmith UI
+3. Note the **Graph ID** from the deployment
+
+**2. Set Required Environment Variables**
+
+```bash
+# Required for all tests
+export LANGSMITH_API_KEY="<your-api-key>"
+export LANGSMITH_WORKSPACE_ID="<your-workspace-id>"
+
+# Required for assistant integration tests
+export TEST_GRAPH_ID="<graph-id-from-deployment>"
+
+# Optional: For specific test scenarios
+export TEST_DEPLOYMENT_ID="<deployment-id>"
+export TEST_ASSISTANT_ID="<existing-assistant-id>"
+```
+
+**Where to find these values:**
+
+- **LANGSMITH_API_KEY**: https://smith.langchain.com/settings → "API Keys"
+- **LANGSMITH_WORKSPACE_ID**: LangSmith UI → Settings → Workspace ID
+- **TEST_GRAPH_ID**: Deployment details page → "Graph ID" field (after deploying test graph)
+- **TEST_DEPLOYMENT_ID**: Deployment details page (optional)
```

Comment:
again, this is unnecessary, out of date or and misleading information. please review elsewhere.